[Yong Zi Xin] iP

zixin448 opened the pull request:
# Duke

Duke helps you to keep track of your events or tasks and it is :
* __Easy to learn__
* Text-based
* Engaging

Here are some types of tasks you can add to Duke:
1.  Event with venue ('/at')
2. Task with deadline ('/by')

Features 💯 
- [x] Managing task
- [ ] Managing deadline (Coming soon!)
- [ ] Reminders (Coming soon!)

> This is a project and it is **fun**! Go play around
> with the programme and have fun!

What do you need to do 👍
1. Download it from [here](https://github.com/zixin448/ip/releases/tag/A-Jar)
2. Have fun!

This is how my <code>main</code> method looks like:
```json
public static void main(String[] args) throws DukeException {
        new Duke("./duke.txt").run();
}
```

Commit:
Add Gradle support

zixin448 committed:
Level-1

zixin448 committed:
Level-2

zixin448 committed:
Level-3

zixin448 committed:
Level-4

zixin448 committed:
Level-5

zixin448 committed:
Level-6

zixin448 committed:
A-TextUiTesting

zixin448 committed:
Level-7

zixin448 committed:
Level-8

zixin448 committed:
Resolve conflict from merge branch-Level-8

zixin448 committed:
A-MoreOOP

zixin448 committed:
A-Packages

zixin448 committed:
Update storage

zixin448 committed:
A-JavaDoc

zixin448 committed:
Clean code

zixin448 committed:
Update code to follow Java coding standard

zixin448 committed:
Add 'find' command

zixin448 committed:
Merge branch 'branch-A-JavaDoc'

Include javadocs for most of the non-private methods

zixin448 committed:
Change variable name to fit coding standard

zixin448 committed:
Resolve conflict Deadline.java

zixin448 committed:
Resolve conflicts from mergine branch-Level-9

zixin448 committed:
A-JUnit

zixin448 committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2122S2/ip into A-Gradle
Merge branch to set up gradle for project

zixin448 committed:
A-Gradle set up gradle to project

Korochii commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,59 @@
+import duke.*;
```
According to the coding standards, imported classes should always be listed explicitly.

Korochii commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,59 @@
+    }
+
+    public static void main(String[] args) throws DukeException {
+//        new Duke("src/main/data/duke.txt").run();
```
Might want to delete this line (can always refer to past commits if needed)

Korochii commented on src/main/java/duke/AddCommand.java:
```diff
@@ -0,0 +1,33 @@
+     * @return String output from ui
+     */
+    public String execute(TaskList taskList, Ui ui, Storage storage) throws IOException {
+        taskList.addTask(task);
+        storage.writeToFile(taskList);
+        return ui.printTaskIsAdded(task);
```
I like how the code is clean and easily understandable here!

Korochii commented on src/main/java/duke/AddCommand.java:
```diff
@@ -0,0 +1,33 @@
+        this.task = task;
+    }
+
+    /**
+     * Handles adding of task to tasklist, calling of ui method to print outpyt and ypdating storage
+     * @param taskList TaskList has all current tasks
+     * @param ui Ui handles printing to output
+     * @param storage Storage saves tasklist
+     * @return String output from ui
+     */
```
```suggestion
    /**
     * Handles adding of task to tasklist, calling of ui method to print output and updating storage.
     *
     * @param taskList TaskList has all current tasks.
     * @param ui Ui handles printing to output.
     * @param storage Storage saves tasklist.
     * @return String output from ui.
     */
```
For this comment (and other comments as well), there should be a spacing between the description and parameter section. In addition, there should be a punctuation behind each parameter description. 
Other than that, there are small typos for the words "output" and "updating".

Korochii commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,78 @@
+        LocalDate taskByDate;
+
+        try {
+            String[] split = input.split(" /by ");
```
split should be in a plural form as it is an array of strings.

Korochii commented on src/main/java/duke/ListCommand.java:
```diff
@@ -0,0 +1,15 @@
+package duke;
+public class ListCommand extends Command{
```
```suggestion
public class ListCommand extends Command {
```
Missing whitespace between Command and {

Korochii commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+                        "OOPS!!! Please enter in format: event <event> /at <event venue> \n " +
+                                "e.g. event Lesson /at Com1");
+            } else {
+                Task task = Event.setEvent(words[1]);
+                if (task == null) {
+                    throw new DukeDeadlineException("Event not added...");
+                } else {
+                    return new AddCommand(task);
+                }
+            }
+
+        } else if (words[0].equals("deadline")) {
+            if (words.length != 2) {
+                throw new DukeDeadlineException("OOPS!!! Please enter in format: deadline <task> /by <yyyy-mm-dd> \n " +
+                        "e.g. deadline complete project /by 2022-12-24");
+            } else {
+                Task task = Deadline.setDeadline(words[1]);
+                if (task == null) {
+                    throw new DukeDeadlineException("Deadline not added...");
+                } else {
+                    return new AddCommand(task);
+                }
+            }
+
+        } else if (words[0].equals("delete")) {
+            int ranking = Integer.parseInt(words[1]);
+            return new DeleteCommand(ranking);
+
+        } else if (words[0].equals("find")) {
+            try {
+//                LocalDate localdate = LocalDate.parse(words[1]);
+                return new FindCommand(words[1]);
+            } catch (Exception e) {
+                throw new DukeException("I'm not sure what's happening. Please try again later!");
+            }
+
+        } else {
+            throw new DukeException("Invalid Command!! Try something else hehe:)");
+        }
+    }
```
I think we should avoid deep nesting here (especially since there is up to 3 levels of if-else statements in line 110).
Perhaps switch statements can be used to compare/parse the first word? :thinking:

Korochii commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,140 @@
+
+    /**
+     * Constructor
+     * Create a storage file in given filepath
```
```suggestion
     * Creates a storage file in given filepath
```
The first sentence should start with creates instead of create

Korochii commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,140 @@
+     * @throws DukeException
+     */
+    public ArrayList<Task> getAllTasks() throws IOException, DukeDeadlineException, DukeException {
+        ArrayList<Task> all = new ArrayList<Task>();
```
```suggestion
        ArrayList<Task> allTasks = new ArrayList<Task>();
```
all should be plural in this case, as it is a collection of tasks

Korochii commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+     * @param ranking
+     */
+    public void markUndone(int ranking) {
+        allTasks.get(ranking-1).unMarkDone();
```
```suggestion
        allTasks.get(ranking - 1).unMarkDone();
```
Missing whitespaces in front/hehind the "-" operator

Korochii reviewed (commented):
Overall, I enjoyed reading your code as the implementation is slightly different from mine (such as the use of Command classes as well as assert statements). However, there are some coding standard violations that can be easily fixed (especially if you decide to work on A-CheckStyle).

Good effort so far and keep up the good work! :smiley:

yokester98 commented on src/main/java/duke/AddCommand.java:
```diff
@@ -0,0 +1,33 @@
+public class AddCommand extends Command {
+    private Task task;
+
+    /**
+     * Constructor
+     * Initialise task attribute to the input task to be added
+     * @param task Task to be added
+     */
```
```suggestion
    /**
     * Constructor
     * Initialise task attribute to the input task to be added
     *
     * @param task Task to be added
     */
```

Good job on all the Javadoc! However, do take note that it is standard convention to leave an empty line between the description field and parameter/return/throw fields.

yokester98 commented on src/main/java/duke/ByeCommand.java:
```diff
@@ -0,0 +1,20 @@
+package duke;
+/**
```
```suggestion
package duke;

/**
```

Overall you are very consistent in leaving an empty line between your package/import statements and your Class statement. Just one minor change here to be made!

yokester98 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,78 @@
+        } catch (Exception e) {
+            DukeDeadlineException error = new DukeDeadlineException(
+                                                    "OOPS!!! Please enter in format: deadline <task> /by <yyyy-mm-dd> \n " +
+                                                    "e.g. deadline complete project /by 2022-12-24");
```
```suggestion
            DukeDeadlineException error = new DukeDeadlineException(
                                                    "OOPS!!! Please enter in format: deadline <task> /by <yyyy-mm-dd> \n "
                                                    + "e.g. deadline complete project /by 2022-12-24");
```

Good job placing line breaks to improve readability! I feel that you should try to break before an operator here.

yokester98 commented on src/main/java/duke/DeleteCommand.java:
```diff
@@ -0,0 +1,30 @@
+     * @throws IOException
+     */
+    public String execute(TaskList tasklist, Ui ui, Storage storage) throws IOException {
+        assert ranking > 0 && ranking <= tasklist.getLength() : "Please give a valid input (0 < input < number of tasks";
```
Good job handling this delete method! I really like how your program can check and remind the user to give a valid input.

yokester98 commented on src/main/java/duke/DeleteCommand.java:
```diff
@@ -0,0 +1,30 @@
+package duke;
+import java.io.IOException;
```
```suggestion
package duke;

import java.io.IOException;
```

Again just nitpicking here, but I feel that there should be a line break between the package statement and import statement.

yokester98 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+     */
+    public static Command parse(String input) throws IOException, DukeException, DukeDeadlineException, DukeEventException {
+        String[] words = input.split(" ", 2);
+        if (words[0].equals("bye")) {
```
You may consider utilising switch statement here, instead of many if-else statements.

yokester98 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+
+
+    /**
+     * Parse all given String input to determine respective corresponding commands that the user is calling
```
```suggestion
     * Parses all given String input to determine respective corresponding commands that the user is calling
```

In method header comments, the first sentence should start in the form Parses..., Sends ..., Adds ... (not Parse or Parsing etc.)

yokester98 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+    public void delete(int ranking) {
+        assert ranking > 0 && ranking <= allTasks.size():
+                "Please give a valid input (0 < input < number of tasks";
+        allTasks.remove(ranking-1);
```
```suggestion
        allTasks.remove(ranking - 1);
```

Operators should be surrounded by a space character.

yokester98 reviewed (commented):
In general, you follow the coding standard very well and your code is very readable, such that I had to nitpick on certain minor issues. Keep up the great work! 👍

zixin448 committed:
Set up JavaFX

zixin448 committed:
Add deadline check

New deadline event should be created only if deadline is after 'today'
Add a new check such that if deadline is before today, system prints
a warning message
Safe guard for consistent values for deadline

zixin448 committed:
Attempt adding assert to deadline

Add assert to deadline event file but received linkage error

zixin448 committed:
Add first assert to ensure deadline is after today and fix bug in src/main/java/duke

1. Structure did not result in printing of actual statements
2. Redirected Event Deadline and FindCommand to the parser
3. Proper output statements are printed
4. Add assert statement to replace original if-else check

zixin448 committed:
Add assert to ensure index input for markCommand is valid

If index input for MarkCommand is larger than store list length or
smaller than 0, will throw indexOutOfBoundException
Add assert statement to check for valid index
Now outputs a warning when given an invalid index

zixin448 committed:
Add assert to unmark and delete commands

zixin448 committed:
Add assertion to commands - Delete, Mark, Unmark, Deadline

Assertion is required to check valid index input of 0<index<=length of task list
and valid deadline input
Add assertion that returns warning to the user
User now sees error message output

zixin448 committed:
Update on code quality

Update file to abide to code quality

zixin448 committed:
Made changes

zixin448 committed:
Merge branch 'master' into branch-A-Assertions

zixin448 committed:
Merge pull request #1 from zixin448/branch-A-Assertions

Branch a assertions

zixin448 committed:
Fix bug when merging

zixin448 committed:
Resolve conflict when merge master into branch

zixin448 committed:
Merge pull request #2 from zixin448/branch-A-CodeQuality

Branch a code quality

zixin448 committed:
Add reminder feature

Reminder is sent out at the start of the chat

zixin448 committed:
Update user guide and GUI

zixin448 committed:
Update README
